llvm/lib/Transforms/Utils/CodeLayout.cpp

```diff
@@ -507,43 +507,56 @@ struct MergedChain {
 
   const NodeT *getFirstNode() const { return *Begin1; }
 
+  bool empty() const { return Begin1 == End1; }
+
+  void append(ObjIter Begin, ObjIter End) {
```

Comment:
This function is introduced just because we're reusing this struct for Jumps (which can only contain up to 2 ranges of jumps).